### PR TITLE
fixing problem with transforming crs for requesting time stamps. see …

### PR DESCRIPTION
**Pull request by AliceBalfanz**
fixing problem with transforming crs for requesting time stamps. see investigation documentation in issue 
https://github.com/dcs4cop/xcube-sh/issues/60#issuecomment-886540083

### CODE REVIEW

**Comment by AliceBalfanz** [comment]
Only failing test is https://github.com/dcs4cop/xcube-sh/blob/master/test/test_reprod_27.py 

therefor I am reopening the related issue: 
https://github.com/dcs4cop/xcube-sh/issues/27

File: xcube_sh/sentinelhub.py [hunk 1]
```diff
@@ -184,7 +184,14 @@ def get_features(self,
             if not source_crs.is_geographic:
                 x1, y1, x2, y2 = bbox
                 transformer = pyproj.Transformer.from_crs(source_crs, 'WGS84')
-                (x1, x2), (y1, y2) = transformer.transform((x1, x2), (y1, y2))
+                # True if CRS is in geocentric (x/y) coordinates, whereas WGS84 is geodetic (y/x)
```
Comment (TonioF) [inline xcube_sh/sentinelhub.py:187]: Please heed maximum line number of 80 characters, also for docs and comments.

File: xcube_sh/sentinelhub.py [hunk 2]
```diff
@@ -184,7 +184,14 @@ def get_features(self,
...
+                # (https://www.w3.org/2015/spatial/wiki/Coordinate_Reference_Systems#EPSG_database)
+                # if source_crs.is_geocentric:
+                # however somehow this does not work, that's why we check whether unit name is degree or not.
+                if not source_crs.axis_info[0].unit_name == 'degree':
```
Comment (TonioF) [inline xcube_sh/sentinelhub.py:191]: I think this is a very unfortunate way to do your check. Can't you use 'is_geographic'?

File: xcube_sh/sentinelhub.py [hunk 3]
```diff
@@ -184,7 +184,14 @@ def get_features(self,
...
+                    (y1, y2), (x1, x2) = transformer.transform((x1, x2), (y1, y2))
+                else:
+                    (x1, x2), (y1, y2) = transformer.transform((x1, x2), (y1, y2))
```
Comment (TonioF) [inline xcube_sh/sentinelhub.py:194]: Can you add a test that goes here? I didn't find one.

**Review by TonioF — COMMENTED** [review]
Please see comments.

**Review by TonioF — APPROVED** [review]

**Review by forman — CHANGES_REQUESTED** [review]
Looks good, thanks! Please update `CHANGES.md` accordingly.

**Review by forman — APPROVED** [review]

**Pull request merged**